blocks/library/gallery/block.js

```diff
@@ -56,13 +57,23 @@ class GalleryBlock extends Component {
 	}
 
 	onSelectImage( index ) {
-		return () => {
+		return ( event ) => {
+			// ignore clicks in the editable caption
```

Comment:
Why?

If it's the case that we want to avoid calling `setState` to the already-selected image, should we not do this more generically to test whether the `index` is any different than `state.selectedImage` ?

In either case, a more detailed comment would be helpful.

Reply:
Hi @aduth, I will add a more detailed comment 👍 But the reason of this check is that without it, making changes in the editable selects/un-selects the image, e.g when we try to select text to make it bold we would be selecting/unselecting many times during this process.